qiime2/core/testing/method.py

```diff
@@ -78,3 +78,17 @@ def optional_artifacts_method(ints: list, num1: int, optional1: list=None,
     if num2 is not None:
         result += [num2]
     return result
+
+
+def variadic_input_method(ints: list, int_set: int, nums: int,
```

Comment:
Why doesn't the type annotation in the signature convey the collection type?

```python
from typing import List, Set

def variadic_input_methods(ints: List[list], int_set: Set[int], nums: Set[int], opt_nums: List[int]=None):
    ...
```

Reply:
The only reason is that `typing.List` doesn't pickle well. This can be fixed outside of Python's stdlib, but we've never really sat down to take care of it. (This is also why we haven't bothered with `typing.Union` which would also be **super** handy.)

Reply:
Makes sense, thanks for the reminder!